src/scrapers/visa-cal.ts

```diff
@@ -411,9 +411,9 @@ class VisaCalScraper extends BaseScraperWithBrowser<ScraperSpecificCredentials>
     const startMoment = moment.max(defaultStartMoment, moment(startDate));
     debug(`fetch transactions starting ${startMoment.format()}`);
 
-    const Authorization = await this.getAuthorizationHeader();
     const cards = await this.getCards();
     const xSiteId = await this.getXSiteId();
+    const Authorization = await this.getAuthorizationHeader();
```

Comment:
You can see how `this.getCards` using the `waitUntil` utility to wait until the value is there (or timeout).

You can use the same approach, and to save time, call both `getCards` and `getAuthorizationHeader` with `Promise.all`.

Reply:
@baruchiro thanks for the feedback done.
I found an issue with the base-scraper:
```
      const success = scrapeResult && scrapeResult.success === true;
      await this.terminate(success);
```
The terminate breaks the tests, but I don't think its related to this fix.
It seems to be related to this issues: #952